cadctap/cadctap/core.py

```diff
@@ -406,17 +415,255 @@ def query(self, query, output_file=None, response_format='VOTable',
+        if pro.startswith(CADC_AC_SERVICE):
+            pro = pro[pro.find('?')+1:]
+        prw = permissions['rw-group']
+        if prw.startswith(CADC_AC_SERVICE):
+            prw = prw[prw.find('?')+1:]
+        perm.add_row((permissions['owner'], permissions['public'], pro, prw))
+        result.append(perm)
+        return result
+
+    def set_permissions(self, resource, read_anon=None, read_only=None,
+                        read_write=None):
+        """
+        Set permissions on a resource (schema or table). Only the permissions
+        that are specified (not None) are being updated on the server.
+
+        :param read_anon: True if anonymous reads are allowed, False otherwise
+        :param read_only: Group URI or empty string (clear existing group)
+        :param read_write: Group URI or empty string (clear existing group)
+        :return:
+        """
+        logger.debug('set_permissions on resource {}: read_anon={}, '
+                     'read_only={}, read_write={}'.format(
+                        resource, read_anon, read_only, read_write))
+        if not resource:
+            raise AttributeError("No resource")
+        if read_anon is None and read_only is None and read_write is None:
+            logger.warning('No permissions values passed in set_permissions')
+            return
+        if read_only and not read_only.startswith('ivo://'):
+            raise AttributeError('Expected URI for read group: {}'.
+                                 format(read_only))
+        if read_write and not read_write.startswith('ivo://'):
+            raise AttributeError('Expected URI for write group: {}'.
+                                 format(read_write))
+        params = ''
+        if read_anon is not None:
+            params += 'public={}\n'.format(str(read_anon).lower())
+        if read_only is not None:
+            ro = read_only
+            params += 'r-group={}\n'.format(ro)
+        if read_write is not None:
+            rw = read_write
+            params += 'rw-group={}\n'.format(rw)
+        self._tap_client.post((PERMISSIONS_CAPABILITY_ID, resource),
+                              data=params,
+                              headers={
+                                  'Content-Type': 'text/plain'})
+
+
+class TabularInfo:
+    """
+    Class to store a very simple tabular information to be displayed in tabular
+    format. The table has a name, a description and an arbitrary number of
+    columns (all the text format).
+    """
+    def __init__(self, name, description, columns):
+        """
+        :param name: name of the schema
+        :param description: description of the schema
+        :param columns: name of the columns. Must match the number of
+        tuple fields in rows
+        """
+        self.name = name
+        self.description = description
+        self.columns = columns
+        self._rows = []
+        self._max_col_lengths = []
+        for c in self.columns:
```

Comment:
A bit pedantic but is there any value in initing the max column lengths with the lengths of the column names? Or the case of a long column name and short column values.

Reply:
You got it - that's the case.